processor/agentmetricsprocessor/utils_combine_process_metrics.go

```diff
@@ -0,0 +1,274 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agentmetricsprocessor
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+// The following code performs a translation from metrics that store process
+// information as resources to metrics that store process information as labels.
+//
+// Starting format:
+//
+//    ResourceMetrics         ResourceMetrics               ResourceMetrics
+// +-------------------+ +-----------------------+     +-----------------------+
+// |  Resource: Empty  | |  Resource: Process 1  |     |  Resource: Process X  |
+// +-------+---+-------+ +---------+---+---------+ ... +---------+---+---------+
+// |Metric1|...|MetricN| |MetricN+1|...|MetricN+M|     |MetricN+1|...|MetricN+M|
+// +-------+---+-------+ +---------+---+---------+     +---------+---+---------+
+//
+// Converted format:
+//
+//                             ResourceMetrics
+// +---------------------------------------------------------------------+
+// |                           Resource: Empty                           |
+// +-------+---+-------+----------------------+---+----------------------+
+// |Metric1|...|MetricN|MetricN+1{P1, ..., PX}|...|MetricN+M{P1, ..., PX}|
+// +-------+---+-------+----------------------+---+----------------------+
+//
+// Assumptions:
+// * There is at most one resource metrics slice without process resource info (will raise error if not)
+// * There is no other resource info supplied that needs to be retained (info may be silently lost if it exists)
+// * All metrics with the same name have the same descriptor (process metrics will be merged if the names match)
+
+func combineProcessMetrics(rms pdata.ResourceMetricsSlice) error {
+	// create collection of combined process metrics, disregarding any ResourceMetrics
+	// with no process resource attributes as "otherMetrics"
+	processMetrics, otherMetrics, err := createProcessMetrics(rms)
+	if err != nil {
+		return err
+	}
+
+	// if no non-process metrics were supplied, initialize an empty
+	// ResourceMetrics object
+	if otherMetrics.IsNil() {
+		otherMetrics.InitEmpty()
+		ilms := otherMetrics.InstrumentationLibraryMetrics()
+		ilms.Resize(1)
+		ilms.At(0).InitEmpty()
+	}
+
+	// append the process metrics to the other ResourceMetrics object
```

Comment:
The fact that you reuse the "other" metrics as a resulting set of metrics is a little confusing to me.
Maybe you should create a separate variable `resultMetrics`, which you will initialize as `resultMetrics := otherMetrics` with an explicit comment mentioning that you're going to reuse it?